### DISCUSSION TITLE
Invariant failed: createServerFn must be called with a function that is marked with the 'use server' pragma.

### DISCUSSION THREAD

**Discussion by SeyyedKhandon** [TanStack/router#2434] [Q&A]
This server function:
```
const getHashId = createServerFn("GET", () => storage.hashId);

```

results in : 

```
Invariant failed: createServerFn must be called with a function that is marked with the 'use server' pragma. Are you using the @tanstack/router-plugin/vite ?
```

while it works for:
```
const getHashId = createServerFn("GET", () => {
  return storage.hashId;
});
```

I am curious what is the reason?

**Comment by schiller-manuel** [comment]
this looks like a bug in our compiler.
however, since server functions are currently re-written, let's wait until this is done and the revisit this issue if it still exists.

**Reply by SeyyedKhandon** [reply]
Should we file a bug report for it, or marking this as a bug in case we dont miss it later then?

**Reply by schiller-manuel** [reply]
sure create a GitHub issue and we will follow up there. ideally, create a failing unit test as a PR

**Comment by SeyyedKhandon** [comment]
Bug report :
https://github.com/TanStack/router/issues/2441
